mcptest/mcptest.go

```
@@ -0,0 +1,118 @@
+	clientWriter io.WriteCloser
+
+	logBuffer bytes.Buffer
+
+	transport transport.Interface
+}
+
+// NewServer starts a new MCP server with the provided tools and returns the server instance.
+func NewServer(t *testing.T, tools ...server.ServerTool) *Server {
+	server := NewUnstartedServer(t)
+	server.AddTools(tools...)
+	server.Start()
```

golangci-lint (1.64.8)

Error return value of `server.Start` is not checked

(errcheck)